app/controllers/users.js

```diff
@@ -1,7 +1,16 @@
 const { signUp } = require('../services/users');
+const logger = require('../logger');
+const { databaseError } = require('../errors');
+
+const loggerPath = 'controller:users';
 
 exports.signUp = async (req, res) => {
-  const { body: userData } = req;
-  const createdUser = await signUp(userData);
-  res.status(201).send(createdUser);
+  try {
+    const { body: userData } = req;
+    const createdUser = await signUp(userData);
+    return res.status(201).send(createdUser);
+  } catch (error) {
+    logger.error(`${loggerPath}:signUp:database: ${error.message}`);
+    throw databaseError('database error in SignUp');
```

Comment:
Here you would not have to throw the error again, you should only pass it to the `next(error)` function